src/main/java/com/alex/eshop/eshop/Service/CategoryServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package com.alex.eshop.eshop.Service;
+
+import com.alex.eshop.eshop.Entity.Category;
+import com.alex.eshop.eshop.Repository.CategoryRepository;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Component
+@Transactional
+public class CategoryServiceImpl implements CategoryService{
+    private final CategoryRepository categoryRepository;
+    public CategoryServiceImpl(CategoryRepository categoryRepository){
+        this.categoryRepository = categoryRepository;
+    }
+    @Override
+    public List<Category> getAllCategories() {
+        return categoryRepository.findAll();
+    }
+    public Category getCategory(Long id){
+        Category category = null;
+        Optional<Category> categoryOptional = categoryRepository.findById(id);
+        if (categoryOptional.isPresent()) {
+            category = categoryOptional.get();
+        }
+        else{
+            throw new NullPointerException();
```

Comment:
`NullPointerException` is wrong exception. Create new exception class `NotFoundException` and throw it. Add message to exception, so we know what exactly is not found